vowpalwabbit/conditional_contextual_bandit.cc

```diff
@@ -420,6 +420,17 @@ void learn_or_predict(ccb& data, multi_learner& base, multi_ex& examples)
 #endif
 
   data.has_seen_multi_slot_example = data.has_seen_multi_slot_example || data.slots.size() > 1;
+
+  // If we have not seen more than one slot, we need to check if the user has supplied slot features.
+  // In that case we will turn on CCB slot id/interactions.
+  if (!data.has_seen_multi_slot_example)
+  {
+    // We decide that user defined features exist if there is at least one feature space which is not the constant
+    // namespace.
+    const bool user_defined_slot_features_exist =
+        !data.slots[0]->indices.empty() && data.slots[0]->indices[0] != constant_namespace;
```

Comment:
Does there need to be a check for data.slots.size() > 0?

Reply:
Oof if slots.size() is 0 there are going to be more issues

Reply:
Do we want to support slots.size() == 0 as syntactic sugar for 1 slot, no features? That will make CB = (CCB without Slots) more explicit.

Reply:
I would rather avoid that, it introduces allocation requirement and extra management. If you parse CB dsjson logs right now with CCB turned on you already get automatic upgrading/sugar. (It adds the slot)